source/base/index_set.cc

```diff
@@ -232,7 +232,27 @@ IndexSet::get_view(const size_type begin, const size_type end) const
   return result;
 }
 
-
+std::vector<IndexSet>
+IndexSet::split_by_block(
+  const std::vector<types::global_dof_index> &dofs_per_block,
+  const IndexSet &                            whole_set)
+{
+  std::vector<IndexSet> partitioned;
+  const unsigned int    n_blocks = dofs_per_block.size();
+  partitioned.clear();
+
+  partitioned.resize(n_blocks);
+  types::global_dof_index start = 0;
+  types::global_dof_index sum   = 0;
+  for (unsigned int i = 0; i < n_blocks; ++i)
```

Comment:
We generally prefer range-based for loops. his shouldbe possible with the suggestions below.
```suggestion
  for (const auto n_block_dofs : dofs_per_block)
```